core/state/statedb.go

```diff
@@ -142,6 +142,9 @@ func New(root common.Hash, db Database, snaps *snapshot.Tree) (*StateDB, error)
 	}
 	// End Quorum - Privacy Enhancements
 
+	// Ensure mapping is type of map[common.Hash][]*types.Log
+	var logs sync.Map
```

Comment:
how are we ensuring that this map stays typesafe? is it common practice to implement specific put/get functions so that the map isn't used in its raw form?

Reply:
Map is not type safe, one way around this is to use wrappers 

```
import "sync"

type Map[K comparable, V any] struct {
	m sync.Map
}

func (m *Map[K, V]) Delete(key K) { m.m.Delete(key) }
func (m *Map[K, V]) Load(key K) (value V, ok bool) {
	v, ok := m.m.Load(key)
	if !ok {
		return value, ok
	}
	return v.(V), ok
}
func (m *Map[K, V]) LoadAndDelete(key K) (value V, loaded bool) {
	v, loaded := m.m.LoadAndDelete(key)
	if !loaded {
		return value, loaded
	}
	return v.(V), loaded
}
func (m *Map[K, V]) LoadOrStore(key K, value V) (actual V, loaded bool) {
	a, loaded := m.m.LoadOrStore(key, value)
	return a.(V), loaded
}
func (m *Map[K, V]) Range(f func(key K, value V) bool) {
	m.m.Range(func(key, value any) bool { return f(key.(K), value.(V)) })
}
func (m *Map[K, V]) Store(key K, value V) { m.m.Store(key, value) }
```

Reply:
We can keep the non-generic code for this `statedb.go`, if we use `s.logs` in other places, then we can consider wrapping it via generics

Reply:
If we start implementing this map in multiple places, then yes, we can use a wrapper to abstract the boxing/unboxing of values, or a different library (non standard lib) that does it